### ISSUE TITLE
Loss Function and Weight Init. Selection

### ISSUE THREAD

**Issue by paramdesai321** [GraceG1225/NeuraSectProj#32]
We need to have the mechanism to select weight init and loss functions.

**Comment by paramdesai321** [comment]
@GraceG1225 Implementation for Loss function and weight init. selection is almost identical to that of Optimizers.

**Comment by paramdesai321** [comment]
> [@GraceG1225](https://github.com/GraceG1225) Implementation for Loss function and weight init. selection is almost identical to that of Optimizers.

For example this is how you call Standard Normal initializer. 

```python
 tf.keras.initializers.RandomNormal(mean=0.0, stddev=1.0)

```

**Comment by paramdesai321** [comment]
List of Initializers:
1.  Standard Normal initialize
2. Xavier initialization
3. He initialization
4. LeCun initialization

**Comment by paramdesai321** [comment]
Create a new branch for the issue. Run these in your local. 

```
git fetch origin
git checkout loss-function-and-weight-init-selection
```

**Comment by paramdesai321** [comment]
List of Loss Functions: 

- Mean Squared Error: tf.keras.losses.MeanSquaredError()
- Mean Absolute Error: tf.keras.losses.MeanAbsoluteError()
- Huber Loss: tf.keras.losses.Huber()
- Binary Cross-Entropy: tf.keras.losses.BinaryCrossentropy()
- Categorical Cross-Entropy: tf.keras.losses.CategoricalCrossentropy()
- Sparse Categorical Cross-Entropy: tf.keras.losses.SparseCategoricalCrossentropy()

**Comment by GraceG1225** [comment]
Adding weight initializer and loss function selection. 6/29/26.

Modification to class Layer:
``` python
_INIT_REGISTRY = {
'standard_normal': tf.keras.initializers.RandomNormal(mean=0.0, stddev=1.0),
'xavier': tf.keras.initializers.GlorotUniform(),
'he': tf.keras.initializers.HeNormal(),
'lecun': tf.keras.initializers.LecunNormal(),
}
```
Responsible for mapping initializer names to their respective Keras objects. (Please also see new parameter 'initializer' in Layer.__init__ and storage of initializer 'self.initializer')

In Layer.Dense():
``` python
init = self._INIT_REGISTRY.get(self.initializer, self.initializer)
return layers.Dense(self.neurons, activation=self.activation,
kernel_regularizer=self._build_regularizer(),
kernel_initializer=init)
```
For registry lookup.

In class MLP:
```python
_LOSS_REGISTRY = {
'mse': tf.keras.losses.MeanSquaredError(),
'mae': tf.keras.losses.MeanAbsoluteError(),
'huber': tf.keras.losses.Huber(),
'binary_crossentropy':  tf.keras.losses.BinaryCrossentropy(),
'categorical_crossentropy': tf.keras.losses.CategoricalCrossentropy(),
'sparse_categorical_crossentropy': tf.keras.losses.SparseCategoricalCrossentropy(),
}
```

Addition to MLP.compile():
``` python
resolved_loss = self._LOSS_REGISTRY.get(loss, loss)
self.model.compile(optimizer=optimizer, loss=resolved_loss, metrics=metrics or [])
```
Responsible for resolving loss string to Keras object; falls back to passing string directly if not found.

**Comment by paramdesai321** [comment]
Looks good to me
